### PR TITLE
Initial Application Skeleton Structure

### PR DESCRIPTION
**Pull request by alexanderbez**
* Implement skeleton structure with a few types already defined
* Updated `Makefile`

Closes: #440 

<hr>

* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))
* [x] Added appropriate labels to PR (ex. wip, ready-for-review, docs)

### CODE REVIEW

**Comment by alexanderbez** [comment]
@AdityaSripal @jackzampolin lmk your thoughts. So far it's a very bare bones skeleton structure. But it should allow us to start implementing core components. I'll be working on #436 based off of this.

File: types/context.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package types
+
+// AppContext provides the ability for the application to pass around and
+// obtain immutable objects easily. More importantly, it allows for the
+// utilization of the object-capability model in which components gain access
+// to other components for which they truly need.
+type AppContext struct {
```
Comment (AdityaSripal) [inline types/context.go:7]: How will this be used for Ethermint? To pass around ChainConfig and stuff like that?

Comment (alexanderbez) [reply types/context.go:7]: Perhaps. I imagine we can/will leverage the existing SDK context, but we probably will need our own for Ethereum stuff. Lmk your thoughts 👍

File: Makefile [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 PACKAGES=$(shell go list ./... | grep -v '/vendor/')
 COMMIT_HASH := $(shell git rev-parse --short HEAD)
 BUILD_FLAGS = -tags netgo -ldflags "-X github.com/cosmos/ethermint/version.GitCommit=${COMMIT_HASH}"
 DOCKER_TAG = unstable
 DOCKER_IMAGE = tendermint/ethermint
+ETHERMINT_DAEMON_BINARY = emintd
```
Comment (jackzampolin) [inline Makefile:20]: Love the new name! Ty for knocking this out!

File: app/ethermint.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+// EthermintApp implements an extended ABCI application.
+type EthermintApp struct {
+	*bam.BaseApp
+
+	codec  *wire.Codec
+	sealed bool
+
+	// TODO: stores and keys
```
Comment (jackzampolin) [inline app/ethermint.go:19]: Nice!

File: cmd/ethermintcli/main.go [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package main
+
+func main() {
+	// TODO: Implement CLI commands and logic
```
Comment (jackzampolin) [inline cmd/ethermintcli/main.go:4]: Good looking todos! Also like the placeholders.

File: version/version.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package version
+
+// Version contains the application semantic version.
+//
+// TODO: How do we want to version this being that an initial Ethermint has
+// been developed?
+const Version = "0.0.0"
```
Comment (jackzampolin) [inline version/version.go:7]: Also nice! Would love to see how this gets plugged in.

**Review by jackzampolin — APPROVED** [review]
Looks great. I've opened an issue for how to do versioning in this project to start a discussion: https://github.com/cosmos/ethermint/issues/448 Otherwise we should get this merged.

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**